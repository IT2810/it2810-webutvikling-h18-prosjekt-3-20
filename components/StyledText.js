import React from 'react';
import { Text, StyleSheet } from 'react-native';

import Colors from '../constants/Colors';

const textStyles = StyleSheet.create({
  mono: {
    fontFamily: 'space-mono',
  },
  error: {
    color: Colors.noticeColor,
  },
});

/**
 * Simple styled text in mono format
 * */
export const MonoText = props => <Text {...props} style={[props.style, textStyles.mono]}/>;

/**
 * Text used to print errors
 * */
export const ErrorText = props => <MonoText {...props} style={[props.style, textStyles.error]}/>;
