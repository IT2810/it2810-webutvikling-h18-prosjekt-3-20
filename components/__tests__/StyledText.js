import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { MonoText, ErrorText } from '../StyledText';


describe('<MonoText/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('<ErrorText/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ErrorText>Some error text</ErrorText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
