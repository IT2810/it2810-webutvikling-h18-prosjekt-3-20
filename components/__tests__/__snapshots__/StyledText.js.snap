// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText/> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Array [
        undefined,
        Object {
          "color": "#FF5555",
        },
      ],
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Some error text
</Text>
`;

exports[`<MonoText/> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
